src/Hl7.Fhir.Base/Rest/BaseFhirClient.cs

```diff
@@ -460,28 +433,44 @@ public virtual async Task DeleteAsync(Resource resource, CancellationToken? ct =
             await DeleteAsync(resource.ResourceIdentity(Endpoint).WithoutVersion(), ct).ConfigureAwait(false);
         }
 
-        /// <summary>
-        /// Delete a resource
-        /// </summary>
-        /// <param name="resource">The resource to delete</param>
-        [Obsolete("Synchronous use of the FhirClient is strongly discouraged, use the asynchronous call instead.")]
-        public virtual void Delete(Resource resource)
-        {
-            DeleteAsync(resource).WaitNoResult();
-        }
-
         /// <summary>
         /// Conditionally delete a resource
         /// </summary>
         /// <param name="resourceType">The type of resource to delete</param>
         /// <param name="condition">Criteria to use to match the resource to delete.</param>
         /// <param name="ct"></param>
+        [Obsolete("As of R6, conditional deletes will be considered separate operations. Use ConditionalDeleteSingle or ConditionalDeleteMultiple instead.")]
         public virtual async Task DeleteAsync(string resourceType, SearchParams condition, CancellationToken? ct = null)
         {
             if (resourceType == null) throw Error.ArgumentNull(nameof(resourceType));
             if (condition == null) throw Error.ArgumentNull(nameof(condition));
 
-            var tx = new TransactionBuilder(Endpoint).Delete(resourceType, condition).ToBundle();
+            var tx = new TransactionBuilder(Endpoint).ConditionalDeleteSingle(condition, resourceType).ToBundle();
+            await executeAsync<Resource>(tx, new[] { HttpStatusCode.OK, HttpStatusCode.NoContent }, ct).ConfigureAwait(false);
+        }
+        
+        /// <summary>
+        /// Conditionally delete a single resource
+        /// </summary>
+        /// <param name="condition">Criteria to use to match the resource to delete</param>
+        /// <param name="resourceType">The type of resource to delete (optional)</param>
+        /// <param name="versionId">The versionId to be used for the if-match header</param>
+        /// <param name="ct"></param>
+        public virtual async Task ConditionalDeleteSingle(SearchParams condition, string? resourceType = null, string? versionId = null, CancellationToken? ct = null)
```

Comment:
I think it's best practice to name this function should be named "Async" at the end.
https://stackoverflow.com/questions/15951774/does-the-use-of-the-async-suffix-in-a-method-name-depend-on-whether-the-async

We don't do it everywhere, but this seems like a good practice for the Client I think